aws-predictions-tensorflow/src/main/java/com/amplifyframework/predictions/tensorflow/service/TensorFlowTextClassificationService.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.tensorflow.service;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.WorkerThread;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.predictions.PredictionsException;
+import com.amplifyframework.predictions.models.Sentiment;
+import com.amplifyframework.predictions.models.SentimentType;
+import com.amplifyframework.predictions.result.InterpretResult;
+import com.amplifyframework.predictions.tensorflow.adapter.SentimentTypeAdapter;
+import com.amplifyframework.predictions.tensorflow.asset.Loadable;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationDictionary;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationLabels;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationModel;
+
+import org.tensorflow.lite.Interpreter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * An implementation of text classification service using
+ * pre-trained model from TensorFlow Lite.
+ */
+final class TensorFlowTextClassificationService {
+    private static final String SERVICE_KEY = "textClassifier";
+
+    // Percentage multiplier
+    private static final int PERCENT = 100;
+
+    private final TextClassificationModel interpreter;
+    private final TextClassificationDictionary dictionary;
+    private final TextClassificationLabels labels;
+    private final List<Loadable<?, PredictionsException>> assets;
+    private final CountDownLatch loaded;
+
+    private PredictionsException loadingError;
+
+    /**
+     * Constructs an instance of service to perform text
+     * sentiment interpretation using TensorFlow Lite
+     * interpreter.
+     * @param context the Android context
+     */
+    TensorFlowTextClassificationService(@NonNull Context context) {
+        this.interpreter = new TextClassificationModel(context);
+        this.dictionary = new TextClassificationDictionary(context);
+        this.labels = new TextClassificationLabels(context);
+
+        this.assets = Arrays.asList(interpreter, dictionary, labels);
+        this.loaded = new CountDownLatch(assets.size());
```

Comment:
You could probablly even wrap this stuff up into its _own_ `Loadable`, a `Loadable<List<Loadable<?, PredictionsException>>, PredictionsException>`. (Lol, that type.) But basically, a loadable that is able to load a list of loadables.

Reply:
Seems like an overkill to me haha